dask/utils.py

```diff
@@ -714,7 +716,10 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     }
 
     if parsed_path.netloc:
-        inferred_storage_options['host'] = parsed_path.hostname
+        # Parse `hostname` from netloc manually because `parsed_path.hostname`
+        # lowercases the hostname which is not always desirable (e.g. in S3):
+        # https://github.com/dask/dask/issues/1417
+        inferred_storage_options['host'] = parsed_path.netloc.rsplit('@', 1)[-1].rsplit(':', 1)[0]
```

Comment:
Does this also work when username/password are not included?  I think that the way you've set things up here should work fine, but it would be good to include a test for this.